roles/kubernetes/preinstall/tasks/dhclient-hooks-undo.yml

```diff
@@ -0,0 +1,22 @@
+---
+
+# These tasks will undo changes done by kargo in the past if needed (e.g. when upgrading from kargo 2.0.x
+# or when changing resolvconf_mode)
+
+- name: Remove kargo specific config from dhclient config
+  blockinfile:
+    dest: "{{dhclientconffile}}"
+    state: absent
+    backup: yes
+    follow: yes
+    marker: "# Ansible entries {mark}"
+  notify: Preinstall | restart network
+
+- name: Remove kargo specific dhclient hook
+  file: path="{{ dhclienthookfile }}" state=absent
+  notify: Preinstall | restart network
+  when: ansible_os_family != "RedHat"
+
+# We need to make sure the network is restarted early enough so that docker can later pick up the correct system
+# nameservers and search domains
+- meta: flush_handlers
```

Comment:
Just verified this. For some reason, Ansible does not flush handlers by itself in this case. I could not find docs that precisely describe when Ansible flushes these, but it looks like they are only flushed after the play (containing multiple roles) has finished.